src/main/java/uk/gov/ons/ctp/response/collection/exercise/client/impl/ActionSvcRestClientImpl.java

```diff
@@ -200,11 +240,30 @@ public ActionRuleDTO updateActionRule(
               new ParameterizedTypeReference<List<ActionRuleDTO>>() {});
     } catch (HttpClientErrorException e) {
       if (e.getStatusCode() == HttpStatus.NOT_FOUND) {
-        return null;
+        throw new CTPException(
```

Comment:
throwing this exception could be done by the caller, which would make the RestClient implementation more consistent. I think we should use the "return null on 404" pattern everywhere.

Reply:
I would agree... but in this case action service gives back 204 if there are no actionRules found and gives back a 404 if the actionPlan you are trying to get the actionRules for was not found. And since this method is called getActionRulesForActionPlan and you provide the actionPlanId you're expecting there to be an actionPlan every time this method is called.